design-documents/graph-ql/coverage/catalog/compare-list.graphqls

```diff
@@ -0,0 +1,46 @@
+type ComparableItem {
+    productId: ID!
+    name: String!
+    sku: String!
+    priceRange: ProductPriceRange!
+    url: ProductUrl
+    images: [ProductImage]
+    values: [ProductAttribute]
+}
+
+type ComparableIndicator {
+    code: String!
+    title: String!
+}
+
+type CompareList {
+    items: [ComparableItem]
+    indicators: [ComparableIndicator]
+}
+
+type Customer{
+    compareList: CompareList
+}
+
+type Query {
+    getCompareListForCustomer(customerId: ID): CompareList
```

Comment:
In another comment, I've asked why compare lists belong to customers, but if they must belong to customers, then It looks like a customer has only one compare list at any time. If that is correct, then it might be better to add `compare_list` as a property on `Customer`.

Reply:
good point, fixed.

Reply:
The compare list is always specific to the buyer but a buyer not always a registered customer. Here is a small difference between guests and registered customers. The customer compare list is permanent when a guest list should not.

Regarding `compare/{sku1}+{sku2}...+{skun}` we can not do because of respect current functionality. Actually Magento stores list state at server, what probably not ideal, but such change will cause a significant change in behavior that I know we do not want to do now.